### ISSUE TITLE
permision denied on nominatim.so on RedHat/CentOS

### ISSUE THREAD

**Issue by mtmail** [twain47/Nominatim#364]
While debugging https://github.com/twain47/Nominatim/issues/363 I installed CentOS7 (7.1) and RedHat Enterprise 7.2 on Amazon AWS.

With our installation instructions a first import of data exits with

    Stopped table: planet_osm_rels in 0s
    node cache: stored: 3126611(100.00%), storage efficiency: 51.68% (dense blocks: 631, sparse nodes: 2702019), hit rate: 100.00%

    Osm2pgsql took 37s overall
    Functions
    CREATE FUNCTION
    ERROR:  could not access file "/home/centos/Nominatim/module/nominatim.so": Permission denied
    ERROR: pgsql returned with error code (3)
    pgsql returned with error code (3)

The permissions of file and path look alright. SElinux reports

    $ ls -Z /home/centos/Nominatim/module/nominatim.so
    -rwxr-xr-x. centos centos unconfined_u:object_r:lib_t:s0   /home/centos/Nominatim/module/nominatim.so

and the access doesn't seem to be logged in `/var/log/audit.log`.

I'm pretty sure it's SElinux fault.

Tested stable (v2.4.0) and latest from git (pretty much v2.5.0).

**Comment by lonvia** [comment]
That's almost always a missing x permission on /home/centos. Have you tried with the vagrant provision script? It explicitly changes home's permissions and runs fine on CentOS 7.2.

**Comment by mtmail** [comment]
Yes, also tried +r. The postgres user can access and read the file, but during the import I keep getting the error.

It might be related to the user account (`vagrant` vs `centos`) setup. Or SELinux vs Vagrant's guess additions.

It works fine on DigitalOcean with CentOS 7.1 so the odd one is Amazon here. I'm leaving my VMs open for another day or two, otherwise I declare temporary defeat.

**Comment by lonvia** [comment]
Another possibility would be that the partition is mounted without exec rights.

Mind if I have a look at the box?

**Comment by easherma** [comment]
Potentially having a similar problem: 

Import
sh: 1: /home/vagrant/Nominatim/osm2pgsql/: Permission denied

This is using vagrant on a Windows machine. The file path for osm2pgsql doesn't seem to populate correctly, when I point it at the correct directory I get the above error.

**Comment by mtmail** [comment]
It's a different error you see. Mine is strictly how the database (Postgres) is loading a shared library and only certain Linux distribution are affected due to extra strict security setting. Probably always happened but we never noticed.

Which vagrant file did you use, Ubuntu or CentOS? Did the provision work when you started the Vagrant virtual machine? Inside the virtual machine, does executing `/home/vagrant/Nominatim/osm2pgsql/osm2pgsql --version` work? Does it work after `chmod +x <filename>`?

**Comment by easherma** [comment]
I have a osm2pgsql but not a second folder inside there. I'll go ahead and post a separate issue.

**Comment by mtmail** [comment]
[ removing easherma's comments since he said he'll resubmit as separate github issue]

On redhat the best I can come up with is that Postgres is already installed and might have the sqpgsql extension enabled. http://www.postgresql.org/docs/current/static/sepgsql.html
In that case looking at `SECURITY LABEL` might be an option. http://www.postgresql.org/docs/current/static/sql-security-label.html 
I can no longer access my virtual machine and terminated it now. Closing for now.

**Comment by KUTlime** [comment]
I have a same issue on CentOS 7. I used vagrant install file but I forked it because soon I realized it is impossible to install Nominatim with the current version.

I changed the ownership of the build folder to the import user but I am getting this error. In addition, there are some warning about version PostgreSQL. yum has only v9.2.

I used v9.6 at Ubuntu, the import was successful.

Any solution?
(Já = Me in my language)
![2018-02-20_160101](https://user-images.githubusercontent.com/8374320/36431244-5e2be6c4-1657-11e8-8f88-851148800c80.png)
